feat: adds nps component

alejoamiras opened the pull request:
This PR adds the most basic form of NPS. We still need to add the custom dimensions to Matomo, but I have tried it on a custom matomo (not aztec's) and it worked :)

<img width="943" height="389" alt="image" src="https://github.com/user-attachments/assets/542d3611-3dd9-4199-a18c-7051c6615cfb" />

`forceNPS()` on console will pop-up NPS widget!

P.S.: I'll remove the "forceNPS" after this is tested & approved :)

critesjosh commented on l1-contracts/lib/circuits:
i think this is unintentional

alejoamiras replied:
That was super-weird. I just merged `next` into my branch and this came with it 👀

critesjosh reviewed (approved):
approving based on this test on the docs preview. im not sure where to verify the results on the matomo site
<img width="1440" height="860" alt="image" src="https://github.com/user-attachments/assets/43e89181-430a-4882-b048-351af86cc2b8" />

alejoamiras commented:
> approving based on this test on the docs preview. im not sure where to verify the results on the matomo site <img alt="image" width="1440" height="860" src="https://private-user-images.githubusercontent.com/18372439/491267819-43e89181-430a-4882-b048-351af86cc2b8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyMjYzNjAsIm5iZiI6MTc1ODIyNjA2MCwicGF0aCI6Ii8xODM3MjQzOS80OTEyNjc4MTktNDNlODkxODEtNDMwYS00ODgyLWIwNDgtMzUxYWY4NmNjMmI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE4VDIwMDc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzYWZhZjQ2MTk1ZDhmMmYzMmQ3YWVmZjQ0NDBmYmQ1ZGY0NzEzMzE0OTEyOGFjNGVlZmFmNDBmYzllZjE0YjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YNBEK9234KA5qBJXChxXkBKjmNNGvc8g_okmEX2c4Vo">

<img width="617" height="803" alt="image" src="https://github.com/user-attachments/assets/f2f1e84e-8507-4c32-8b15-ba0bb6f4d8bd" />

It did get to our Matomo 🥳  - now... creating the report for getting our NPS is another challenge.

alejoamiras committed:
feat: adds nps component

This PR adds the most basic form of NPS. We still need to add the custom dimensions to Matomo, but I have tried it on a custom matomo (not aztec's) and it worked :)

<img width="943" height="389" alt="image" src="https://github.com/user-attachments/assets/542d3611-3dd9-4199-a18c-7051c6615cfb" />

`forceNPS()` on console will pop-up NPS widget!

P.S.: I'll remove the "forceNPS" after this is tested & approved :)

Co-authored-by: Alejo Amiras <alejo.amiras@gmail.com>
